spk/gerrit/src/installer.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/sh
+
+# Package
+PACKAGE="gerrit"
+DNAME="Gerrit"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+SSS="/var/packages/${PACKAGE}/scripts/start-stop-status"
+JAVA_DIR="/var/packages/JavaManager/target/Java/jre"
+GIT_DIR="/usr/local/git"
+PATH="${INSTALL_DIR}/bin:${GIT_DIR}/bin:${JAVA_DIR}/bin:/usr/local/bin:/bin:/usr/bin:/usr/syno/bin"
+USER="gerrit"
+GROUP="users"
+GIT="${GIT_DIR}/bin/git"
+
+SERVICETOOL="/usr/syno/bin/servicetool"
+FWPORTS="/var/packages/${PACKAGE}/scripts/${PACKAGE}.sc"
+
+BOUNCYCASTLE_VER="151"
+BOUNCYCASTLE_PROVIDER_URL="http://www.bouncycastle.org/download/bcprov-jdk15on-${BOUNCYCASTLE_VER}.jar"
+BOUNCYCASTLE_SSL_URL="http://www.bouncycastle.org/download/bcpkix-jdk15on-${BOUNCYCASTLE_VER}.jar"
+
+#SYNO_GROUP
+#SYNO_GROUP_DESC
+
+DEFAULT_PLUGINS="download-commands review"
+
+preinst ()
+{
+	exit 0
+}
+
+postinst ()
+{
+	# Link
+	ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+	# Install busybox stuff
+	${INSTALL_DIR}/bin/busybox --install ${INSTALL_DIR}/bin
+
+	if [ "${SYNOPKG_PKG_STATUS}" != "UPGRADE" ]; then
+		# Create user
+		adduser -h ${INSTALL_DIR} -g "${DNAME} User" -G ${GROUP} -s /bin/sh -S -D ${USER}
+
+		# Install prerequsite libraries
+		wget -nv -c "$BOUNCYCASTLE_PROVIDER_URL" -O "${SYNOPKG_PKGDEST}/lib/bcprov-jdk15on-${BOUNCYCASTLE_VER}.jar"
+		wget -nv -c "$BOUNCYCASTLE_SSL_URL" -O "${SYNOPKG_PKGDEST}/lib/bcpkix-jdk15on-${BOUNCYCASTLE_VER}.jar"
+
+		# Correct the file ownership
+		chown -R ${USER}:${GROUP} ${SYNOPKG_PKGDEST}
+
+		# Install Gerrit and initialise secondary index (if upgrading, do this after restoring working data in postupgrade)
+		su - ${USER} -c "PATH=${PATH} java -jar ${SYNOPKG_PKGDEST}/bin/gerrit.war init --batch --no-auto-start --site-path=${SYNOPKG_PKGDEST} --install-plugin=download-commands --install-plugin=reviewnotes --install-plugin=singleusergroup --install-plugin=replication"
+		su - ${USER} -c "PATH=${PATH} java -jar ${SYNOPKG_PKGDEST}/bin/gerrit.war reindex --site-path=${SYNOPKG_PKGDEST}"
+
+		# Change pid check
+		sed -i "s#ps -p \$PID#ls /proc/\$PID#" ${SYNOPKG_PKGDEST}/bin/gerrit.sh
+
+		# Add firewall config
+		${SERVICETOOL} --install-configure-file --package ${FWPORTS} >> /dev/null
+	fi
+
+	# Only correct the file ownership
+	chown -R ${USER}:${GROUP} ${SYNOPKG_PKGDEST}
+
+	exit 0
+}
+
+preuninst ()
+{
+	# Stop the package
+	${SSS} stop > /dev/null
+
+	# Remove the user (if not upgrading)
+	if [ "${SYNOPKG_PKG_STATUS}" != "UPGRADE" ]; then
+		delgroup ${USER} ${GROUP}
+		deluser ${USER}
+	fi
+
+	# Remove firewall config
+	if [ "${SYNOPKG_PKG_STATUS}" == "UNINSTALL" ]; then
+		${SERVICETOOL} --remove-configure-file --package ${PACKAGE}.sc >> /dev/null
+	fi
+
+	exit 0
+}
+
+postuninst ()
+{
+	# Remove link
+	rm -f ${INSTALL_DIR}
+
+	exit 0
+}
+
+preupgrade ()
+{
+	# Stop the package
+	${SSS} stop > /dev/null
+
+	# Save all the working data
+	rm -fr ${TMP_DIR}/${PACKAGE}
+	mkdir -p ${TMP_DIR}/${PACKAGE}
+	for D in data db etc git index lib logs plugins; do
```

Comment:
The installation of Gerrit is greatly simplified by dumping everything into what it calls the $SITE_PATH which is then effectively managed by the 'init' call (e.g. an upgrade is handled by calling 'init' on an existing install directory).

I'll see how much of this can be configured, but I've left it as-is for now.